scripts/smoke-test.js

```diff
@@ -22,13 +22,13 @@ function uncommentPythonCodeBlocks(content) {
     .split(/\r?\n/)
     .map((line) => {
       if (insidePythonCodeBlock) {
-        line = line.replace(/^      \#/, '      ');
-        quotationMarksSeen += /^      """/.test(line) ? 1 : 0;
+        line = line.replace(/^ {6}#/, '      '); // eslint-disable-line no-param-reassign
```

Comment:
Why to mutate a string when we are in the `.map()`?
Let's return a new string, applying necessary transformations (replace, etc.).

Reply:
It's not really that much about mutation as it is about keeping the code easy to read. Compare to this:

```js
    .map((line) => {
      let processedLine;
      if (insidePythonCodeBlock) {
        processedLine = line.replace(/^ {6}#/, '      ');
        quotationMarksSeen += /^ {6}"""/.test(line) ? 1 : 0;
        insidePythonCodeBlock = quotationMarksSeen < 2;
      } else {
        processedLine = line;
        quotationMarksSeen = 0;
        insidePythonCodeBlock = /(Given|And) a file named ".+\.py"/i.test(line);
      }
      return processedLine;
    })
```

Only to avoid reassigning to the function argument 😕

Reply:
It's okay. There is no need to focus on code refactoring as a part of linting introduction. I leave this up to you.